subprojects/language-native/src/main/java/org/gradle/language/swift/tasks/SwiftCompile.java

```diff
@@ -294,11 +296,21 @@ public void execute(InputFileDetails removed) {
                     removedFiles.add(removed.getFile());
                 }
             });
+
+            // If a non-source file changed, rebuild everything.
+            if (!changedFiles.isEmpty()) {
+                Set<File> allSourceFiles = getSource().getFiles();
+                if (!allSourceFiles.containsAll(changedFiles)) {
+                    changedFiles.clear();
+                    changedFiles.addAll(allSourceFiles);
```

Comment:
This whole conditional feels a little clunky.  Seems like instead of this `if (incremental) /else` statement, we should have a `SwiftIncrementalCompiler` that looks at the incremental flag on the spec and does the cleaning if its not incremental.  Then here, if we have a changed non-source file, we just mark the spec not incremental.